src/datapoints-graph/charts/charts.component.ts

```diff
@@ -143,6 +162,242 @@ export class ChartsComponent implements OnChanges, OnInit, OnDestroy {
+              return;
+            } else {
+              const seriesValue = this.findValueForExactOrEarlierTimestamp(
+                series.data,
+                XAxisValue
+              );
+              if (!seriesValue) {
+                return;
+              }
+              value =
+                seriesValue[1]?.toString() +
+                (series.datapointUnit ? ` ${series.datapointUnit}` : '') +
+                `<div style="font-size: 11px">${this.datePipe.transform(
+                  seriesValue[0]
+                )}</div>`;
+            }
+
+            YAxisReadings.push(
+              `<span style='display: inline-block; background-color: ${series.itemStyle.color} ; height: 12px; width: 12px; border-radius: 50%; margin-right: 4px;'></span>` + // color circle
+                `<strong>${series.datapointLabel}: </strong>` + // name
+                value // single value or min-max range
+            );
+          });
+
+          const event = this.events.find(
+            (e) => (e.type = params.data.eventType)
+          );
+          const alarm = this.alarms.find(
+            (a) => (a.type = params.data.alarmType)
+          );
+
+          let value: string;
+          if (event) {
+            // Add the event information to the value
+            value = `<div style="font-size: 11px">Event Time: ${event.time}</div>`;
+            value += `<div style="font-size: 11px">Event Type: ${event.type}</div>`;
+            value += `<div style="font-size: 11px">Event Text: ${event.text}</div>`;
+            value += `<div style="font-size: 11px">Event Last Updated: ${event.lastUpdated}</div>`;
+          }
+
+          if (alarm) {
+            // Add the alarm information to the value
+            value = `<div style="font-size: 11px">Alarm Time: ${alarm.time}</div>`;
+            value += `<div style="font-size: 11px">Alarm Type: ${alarm.type}</div>`;
+            value += `<div style="font-size: 11px">Alarm Text: ${alarm.text}</div>`;
+            value += `<div style="font-size: 11px">Alarm Last Updated: ${alarm.lastUpdated}</div>`;
+          }
+          YAxisReadings.push(value);
+          return (
+            this.datePipe.transform(XAxisValue) +
+            '<br/>' +
+            YAxisReadings.join('')
+          );
+        };
+        this.echartsInstance.setOption(updatedOptions);
+      }
+    });
+
+    this.echartsInstance.on('mouseout', () => {
+      const options = this.echartsInstance.getOption();
+      if (originalFormatter) {
+        options.tooltip[0].formatter = originalFormatter;
+        this.echartsInstance.setOption(options);
+      }
+    });
+  }
+
+  onChartClick(params) {
+    if (this.isAlarmClick(params)) {
+      const clickedAlarms = this.alarms.filter(
+        (alarm) => alarm.type === params.data.alarmType
+      );
+
+      const options = this.echartsInstance.getOption();
+
+      const timeRange = this.getTimeRange();
+      const updatedOptions = !this.hasMarkArea(options)
```

Comment:
updated options also seems like something to separate, I'm leaning towards some service as charts.component grown almost twice with this PR...